index.js

```diff
@@ -1,32 +1,29 @@
 /* eslint-env node */
 'use strict';
 
-var path = require('path');
-var filterInitializers = require('fastboot-filter-initializers');
-var VersionChecker = require('ember-cli-version-checker');
-var mergeTrees = require('broccoli-merge-trees');
+const path = require('path');
+const VersionChecker = require('ember-cli-version-checker');
+const mergeTrees = require('broccoli-merge-trees');
 
 module.exports = {
   name: 'ember-cli-head',
 
-  treeForApp: function(defaultTree) {
-    var checker = new VersionChecker(this);
-    var emberVersion = checker.for('ember-source', 'npm');
+  treeForApp(defaultTree) {
+    let checker = new VersionChecker(this);
```

Comment:
My PR exposes a process environment `process.env.FASTBOOT_NEW_BUILD` and `addToFastBootTree` for these cases. Instead of merging here you can just do this:

Let me know if this doesn't work or you feel this may not be needed but I exposed those hooks in my PR to have this in mind. More than happy to drop code 😄 

```js
treeForApp(defaultTree) {
  if (!process.env.FASTBOOT_NEW_BUILD) {
      var emberVersion = checker.for('ember-source', 'npm');

    if (!emberVersion.version) {
      emberVersion = checker.for('ember', 'bower');
    }

    var trees = [defaultTree];

    // 2.9.0-beta.1 - 2.9.0-beta.5 used glimmer2 (but 2.9.0 did not)
    // 2.10.0-beta.1+ includes glimmer2
    if (!(emberVersion.gt('2.9.0-beta') && emberVersion.lt('2.9.0')) && !emberVersion.gt('2.10.0-beta')) {
      trees.push(this.treeGenerator(path.resolve(this.root, 'app-lt-2-10')));
    }

    var tree = mergeTrees(trees, { overwrite: true });

    return filterInitializers(tree);
  } else {
     return defaultTree;
 }
}

addToFastBootTree() {
  var fastbootBuilder = require('ember-cli-fastboot/lib/build-utilities/fastboot-builder');
  
  let checker = new VersionChecker(this);
  let emberVersion = checker.for('ember-source', 'npm');
 
  if (!emberVersion.version) {
     emberVersion = checker.for('ember', 'bower');
   }

  // 2.9.0-beta.1 - 2.9.0-beta.5 used glimmer2 (but 2.9.0 did not)
    // 2.10.0-beta.1+ includes glimmer2
    if (!(emberVersion.gt('2.9.0-beta') && emberVersion.lt('2.9.0')) && !emberVersion.gt('2.10.0-beta')) {
      return fastbootBuilder.addFastBootPath(path.resolve(this.root, 'fastboot-app-lt-2-9'));
    }
}